src/Microsoft.DotNet.Web.ProjectTemplates/content/WebApi-FSharp/Controllers/ValuesController.fs

```diff
@@ -13,11 +13,13 @@ type ValuesController () =
 
     [<HttpGet>]
     member this.Get() =
-        [|"value1"; "value2"|]
+        let values = [|"value1"; "value2"|]
+        ActionResult<string[]>(values)
```

Comment:
@cartermp If someone wanted to return Found or some other ActionResult will they replace `ActionResult` here? I am not sure I understand how this guides anyone using the template to use the recommended pattern @Eilon mentioned [here](https://github.com/aspnet/templating/pull/351#issuecomment-372380792)

Reply:
Hmm, I've only ever done ASP.NET MVC without explicit ActionResults in F#, which makes things a lot easier. Doing it explicitly like this makes it awkward, since you'll need to have all return paths casted to a base class or interface (not sure which would be best for MVC).

Making this template a pathway to learning that would be a challenge due to the lack of implicit conversions. I don't think it's particularly harmful to have these explicitly right now, but I think a lot of F# developers will just go back to returning the data as-is.

Reply:
@jbagga I think that if you start from this C#:

```c#
if (blah)
{
    return new string[] { "one", "two" };
}
else
{
    return NotFoundResult(); // helper factory method
}
```

Then after the explicit conversions, this code effectively ends up being:

```c#
if (blah)
{
    return new ActionResult<string[]>(new string[] { "one", "two" });
}
else
{
    return new ActionResult<string[]>(NotFoundResult()); // helper factory method
}
```

And in C# because the declared return type would be `ActionResult<string[]>`, the MVC metadata system can deduce many things about the result.

So in F#, though I don't know what the code would exactly look like, I hope you can do something like this totally made up code:

```f#
        blah ??
            if true ActionResult<string[]>(values)
            if false ActionResult<string[]>(this.NotFoundResult)
```

And then if the F# compiler figures out that the declared return type is `ActionResult<string[]>`, then I think everything will work.

Reply:
@Eilon yep!

```fsharp
    [<HttpGet("{id}")>]
    member this.Get(id:int) =
        if id > 100 then
            ActionResult<string>("Over 100!")
        else
            ActionResult<string>(NotFoundResult())
```

And this works as you'd expect - I get a 404 not found if `id` is 100 or less. I didn't actually know that you can pass `NotFoundResult` into an `ActionResult`. Fun times.

Reply:
Ok then this looks good! Thanks @Eilon and @cartermp. Approving the PR now